src/test/ui/pattern/issue-88074-pat-range-type-inference-err.stderr

```diff
@@ -0,0 +1,12 @@
+error[E0029]: only `char` and numeric types are allowed in range patterns
+  --> $DIR/issue-88074-pat-range-type-inference-err.rs:11:9
+   |
+LL |         Zero::ZERO ..= Zero::ZERO => {},
+   |         ----------^^^^^----------
+   |         |              |
+   |         |              this is of type `_` but it should be `char` or numeric
+   |         this is of type `_` but it should be `char` or numeric
```

Comment:
Somewhere we are not actually using `resolve_ty_if_possible`. We should be mentioning that these are of type `String`.